web/src/site-admin/SiteAdminRepositoriesPermissionsPage.tsx

```diff
@@ -0,0 +1,133 @@
+                <div className="d-flex align-items-center justify-content-between">
+                    <div>
+                        <RepoLink repoName={this.props.node.repository.name} to={this.props.node.repository.url} />
+                    </div>
+                    <div className="repository-node__actions">
+                        <small className="ml-2 text-muted">
+                            <Timestamp date={this.props.node.permissions.updatedAt} />
+                        </small>{' '}
+                    </div>
+                </div>
+                {this.state.errorDescription && <ErrorAlert className="mt-2" error={this.state.errorDescription} />}
+            </li>
+        )
+    }
+}
+
+interface Props extends RouteComponentProps<{}>, ActivationProps {}
+
+class FilteredRepositoryPermissionsConnection extends FilteredConnection<
+    GQL.IRepositoryPermissions,
+    Pick<RepositoryPermissionsNodeProps, 'onDidUpdate'>
+> {}
+
+/**
+ * A page displaying the repositories permissions on this site.
+ */
+export class SiteAdminRepositoriesPermissionsPage extends React.PureComponent<Props> {
+    private static FILTERS: FilteredConnectionFilter[] = [
+        {
+            label: 'Most recent synced',
+            id: 'most-recent-synced',
+            tooltip: 'Sort by most recent synced',
+            args: {},
+        },
+        {
+            label: 'Least recent synced',
+            id: 'least-recent-synced',
+            tooltip: 'Sort by least recent synced',
+            args: { descending: false },
+        },
+    ]
+
+    private repositoryUpdates = new Subject<void>()
+
+    public componentDidMount(): void {
+        eventLogger.logViewEvent('SiteAdminReposPerms')
+
+        // Refresh global alert about enabling repositories when the user visits here.
+        refreshSiteFlags()
+            .toPromise()
+            .then(null, err => console.error(err))
+    }
+
+    public componentWillUnmount(): void {
+        // Remove global alert about enabling repositories when the user navigates away from here.
+        refreshSiteFlags()
+            .toPromise()
+            .then(null, err => console.error(err))
+    }
+
+    public render(): JSX.Element | null {
+        const nodeProps: Pick<RepositoryPermissionsNodeProps, 'onDidUpdate' | 'activation'> = {
+            onDidUpdate: this.onDidUpdateRepository,
+            activation: this.props.activation,
+        }
+
+        return (
+            <div className="site-admin-repositories-page">
+                <PageTitle title="Repositories permissions - Admin" />
+                <h2>Repositories permissions</h2>
+                <p>
+                    Repositories permissions are synced from connected{' '}
+                    <Link to="/site-admin/external-services">code host connections</Link>.
+                </p>
+                <FilteredRepositoryPermissionsConnection
```

Comment:
This avoids bloating the bundle with extra classes
```suggestion
                <FilteredConnection<GQL.IRepositoryPermissions, Pick<RepositoryPermissionsNodeProps, 'onDidUpdate'>>
```